docs/source/miscellaneous/doc/adr/0014-cassandra-targeted-sweep.md

```diff
@@ -0,0 +1,283 @@
+
+  **Column components**:
+  - **timestamp_modulus**: a `VAR_LONG` storing the start timestamp of the write modulo 50_000.
+  - **write_index**: a `VAR_SIGNED_LONG` whose purpose is overloaded. In case we are storing 50 or fewer writes into the same row,
+    the *write_index* is a non-negative increasing number used to deduplicate multiple writes with the same start timestamp. If we
+    are storing more writes, we instead only put a single cell in the non-dedicated row acting as a reference to one or more
+    dedicated rows. In this case, the *write_index* is a negative number between 1 and 64 indicating how many dedicated rows we are
+    using (each dedicated row can contain up to *100_000* cells). In dedicated rows, the *write_index* is once again used for deduplication since all entries in the row will have the same
+    *timestamp_modulus*.
+
+  **Value**: a `WriteReference` containing the remaining required metadata for targeted sweep: the `TableReference`, `Cell`, and
+    a `boolean` specifying if the write was a tombstone.
+
+  Since a persisted size of a `Cell` object can be up to 3_000 bytes (dominating the size of the above entry), and we can have at
+  most 12_500 transactions per row, by allowing at most 50 writes
+  from the same transaction in a row, we ensure that the size of one non-dedicated row will not exceed 2 GB and should in practice
+  be lower than 100 MB. We allow a maximum of 100_000 entries per dedicated row, which ensures that each dedicated row should not be
+  larger than 300MB while still allowing up to 6.4 million writes in a single transaction, and in practice even more, as long as
+  the number of shards is greater than 1. Note that for each cell in a non-dedicated row, we can calculate the write's start
+  transaction simply by multiplying the *timestamp_partition* by *50_000* and then adding its *timestamp_modulus*.
+
+#### sweepableTimestamps
+
+  This is an auxiliary table for locating the next row of the *sweepableCells* table to read since the timestamp partitions can be
+  sparse, and therefore requiring many lookups to locate a nonempty row. Each non-dedicated row of *sweepableCells* is represented
+  by a single cell in this table.
+
+  **Row components**:
+  - **shard**: a `VAR_LONG` containing the shard for which the row has entries for.
+  - **timestamp_partition**: a `VAR_LONG` corresponding to the *coarse timestamp partition* of all entries in the row.
+  - **sweep_conservative**: a `boolean` (encoded as a `BLOB`) specifying if the row contains entries for thorough (`0x00`) or
+    conservative sweep (`0x01`).
+
+  **Column components**:
+  - **timestamp_modulus** is the **fine timestamp partition** of a row in SweepableCells that falls into the coarse partition
+    specified in the timestamp_partition row component.
+
+  **Value**: unused empty byte array.
+
+  Locating the next row of *sweepableCells* to read is then done by reading the next column in the coarse partition we are currently
```

Comment:
I think row is the wrong word here, since you mention iterating to the next column if we're already in a fine partition. The first part mentions the coarse partition, suggesting we're talking about sweepableTimestamps, but the second section suggests we weren't (instread) - something is a bit confusing here.

Reply:
Reworded, hopefully clearer now